pkg/parser/generate.go

```diff
@@ -0,0 +1,3 @@
+package parser
+
+//go:generate pigeon -optimize-parser -alternate-entrypoints AsciidocRawDocument,RawFile,TextDocument,DocumentRawBlock,FileLocation,IncludedFileLine,InlineLinks,LabeledListItemTerm,NormalBlockContent,NormalParagraphContent,VerseBlockContent,MarkdownQuoteBlockAttribution,InlineElements -o parser.go parser.peg	
```

Comment:
Ok, I see. That's actually what I was missing in the earlier version of the PR ;) 
In this case, we should update the `generate-optimized` goal in https://github.com/bytesparadise/libasciidoc/blob/master/make/go.mk#L40-L46 to call `go generate ./...` instead, so we don't have 2 places where list of `alternate-entrypoints` need to be maintained.

(Note that for now I'd like to keep the `make generate` goal because I use it in my daily testing workflow: it keeps my code/generate/test workflow pretty fast.)

Reply:
Sure -- but this also helps me ... I always have to shell out to WSL to generate the grammar because I don't have a version of "make" that understands your makefiles on Windows.  (Your makefiles are GNU make, and not stock simple make, which adds yet further onus on folks who have to use them.)

Do you have any use for a separate generate rule, or do you think we could just consolidate those to have and generate and generate-optimized both generate the optimized parser?

Reply:
As I said:
1. I understand it can be complicated for you on Windows, so I'm fine with using `go generate ./...` under the hood.
2. This PR should refactor the `generate-optimized` goal to use the `go generate ./...` command
3. For now, I'm used to `make generate` because it's faster in my workflow (code/generate/test) so I would like to keep this specific goal as-is. However, I always run `make test` before submitting a PR anyways, so the parser will be optimized at this point.

Does it clarify?

Reply:
Yes, thanks.  I'm surprised you can tell the difference in speed between generate and generate-optimized.  I haven't measured it, but its far below my threshold of notice.

Reply:
Hmm, I'll have to check again then, but that was my feeling... 
Anyways, I believe that your main request is being able to use `go generate` and mine is to make sure we don't have to maintain the `alternate-entrypoints` in 2 different places. Once we have the `make generate-optimized` goal using `go generate` under the hood, I will give it another try and maybe remove the `generate` goal afterwards. 

Also, I wonder what the Windows VM on GitHub Actions use, because the `make` commands work fine there 🤔

Reply:
I think they have a full CYGWIN environment.  Which also means GNU make.

Btw, GNU make is *not* portable make.  That syntax isn't understand by NMake (the Visual studios make), pmake (the FreeBSD make utility), or make(1S) as it is found on illumos (though some distributions ship gnumake instead because a lot of people use GNUisms in their makefiles.)

For a pure Go project, its nice to not require any tooling other than what you can get via go get.  It reduces friction.

(For the record, I do use make-like utilities in some of my other projects, but for those I'm using mostly CMake and ninja, which are very nice but completely inappropriate for a Go project.)

Reply:
Sure, I totally understand your concerns about the friction! 
We use `make` on our projects but we're all on Linux or macOS, so it works in our cases, which is also why I started this project with the tools I'm most familiar with. 
That being said, I'm fine with make your life (and others') easier here, and maybe we can remove all the `make` stuff in the future. Just one thing at a time, ok? ;)

Reply:
> Hmm, I'll have to check again then, but that was my feeling...
> Anyways, I believe that your main request is being able to use `go generate` and mine is to make sure we don't have to maintain the `alternate-entrypoints` in 2 different places. Once we have the `make generate-optimized` goal using `go generate` under the hood, I will give it another try and maybe remove the `generate` goal afterwards.
> 
> Also, I wonder what the Windows VM on GitHub Actions use, because the `make` commands work fine there 🤔

@gdamore last time I checked there were actually barely no difference of execution time between `make generate` and `make generate-optimized`. I'll refactor the `go.mk` later in a subsequent PR